### PR TITLE
BACKLOG-12710 implement basic search modal and its button

### PR DESCRIPTION
**Pull request by DameniMilo**
## JIRA
https://jira.jahia.org/browse/BACKLOG-12710

## Description
implement basic search modal and its button

## Tests

The following are included in this PR
- [ ] Unit Tests (Most changes _should_ have unit tests)
- [ ] Integration Tests

## Checklist
I have considered the following implications of my change: 

- [ ] Security (in particular for changes to authentication, authorization, data fetching, ...)
- [ ] Performance
- [ ] Migration
- [ ] Code maintainability

## Documentation

- [ ] Inline documentation
- [ ] Internal Documentation (wiki)
- [ ] User-facing Documentation

### CODE REVIEW

File: src/javascript/JContent/actions/searchAction/SearchDialog/AdvancedSearch/AdvancedSearch.jsx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+import {Trans} from 'react-i18next';
+import {DxContext} from '@jahia/react-material';
+import {compose} from '~/utils';
+import {cmGoto} from '~/JContent/JContent.redux';
+import JContentConstants from '~/JContent/JContent.constants';
+import styles from './AdvancedSearch.scss';
+import SearchLocation from '../SearchLocation';
+import {Typography} from '@jahia/moonstone';
+
+export class AdvancedSearch extends React.Component {
```
Comment (cedricmoitrier) [inline src/javascript/JContent/actions/searchAction/SearchDialog/AdvancedSearch/AdvancedSearch.jsx]: Could we have this component split in two, with a container and a "pure" UI component?

File: src/javascript/JContent/actions/searchAction/SearchDialog/AdvancedSearch/Sql2Input/Sql2Input.jsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {Input} from '@jahia/design-system-kit';
+
+export class Sql2Input extends React.Component {
```
Comment (cedricmoitrier) [inline src/javascript/JContent/actions/searchAction/SearchDialog/AdvancedSearch/Sql2Input/Sql2Input.jsx]: Why not a functional component?

File: src/javascript/JContent/actions/searchAction/SearchDialog/BasicSearch/BasicSearch.jsx [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        }
+    });
+
+    if (error) {
+        console.log(error);
+    }
+
+    let contentTypeSelectData;
+    let selectedContentType;
+    if (!loading && data && data.jcr && data.jcr.nodeTypes && data.jcr.nodeTypes.nodes) {
```
Comment (cedricmoitrier) [inline src/javascript/JContent/actions/searchAction/SearchDialog/BasicSearch/BasicSearch.jsx]: What about "data?.jcr?.nodeTypes?.nodes" ?

File: src/javascript/JContent/actions/searchAction/SearchDialog/AdvancedSearch/AdvancedSearch.jsx (hunk 1 above)
Comment (DameniMilo) [reply src/javascript/JContent/actions/searchAction/SearchDialog/AdvancedSearch/AdvancedSearch.jsx]: AdvancedSearch is part of another story, I did some changes in it, but I mainly moved the file to right place, in the following story it should be transformed into a functional component as anyway the logic will likely not be compatible with the rest of the UI

File: src/javascript/JContent/actions/searchAction/SearchDialog/AdvancedSearch/Sql2Input/Sql2Input.jsx (hunk 2 above)
Comment (DameniMilo) [reply src/javascript/JContent/actions/searchAction/SearchDialog/AdvancedSearch/Sql2Input/Sql2Input.jsx]: Same as for AdvancedSearch

**Comment by DameniMilo** [comment]
I will create another task for unit test, as for now, I have a modal but some stuff may change since the search is not working

File: src/javascript/JContent/actions/searchAction/SearchDialog/BasicSearch/BasicSearch.jsx (hunk 3 above)
Comment (DameniMilo) [reply src/javascript/JContent/actions/searchAction/SearchDialog/BasicSearch/BasicSearch.jsx]: done

**Pull request merged**